wgpu-types/src/lib.rs

```diff
@@ -7401,6 +7401,9 @@ pub enum Dx12Compiler {
         /// Path to the `dxcompiler.dll` file, or path to the directory containing `dxcompiler.dll` file. Passing `None` will use standard platform specific dll loading rules.
         dxc_path: Option<PathBuf>,
     },
+    /// The statically-linked variant of Dxc, sourced from the [`mach-dxcompiler-rs`](https://crates.io/crates/mach-dxcompiler-rs) crate.
+    /// The `mach-dxcompiler-rs` feature is required to use this.
+    MachDxc,
```

Comment:
I would call this StaticDxc, and maybe rename the other one DynamicDxc. The user doesn't much care about where it comes from, just that it's a statically linked version of DXC.